src/domain-engine.js

```diff
@@ -85,17 +85,30 @@ class DomainEngine {
     return domain
   }
 
-  dispatch(state: Object, action: Action): Object {
+  dispatch(action: Action, state: Object, snapshot: Object) {
```

Comment:
Yeah. Exactly. The handlers are cached prior invocations of `register()` on domains so that we don't have to call them every time we dispatch an action.

The handlers are what actually modify state, like the value part of register:

```javascript
const Domain = {
  register() {
    return {
      [action]: handler
    }
  }
}
```